src/controller/jobmonitor/monitor.go

```diff
@@ -0,0 +1,142 @@
+//  Copyright Project Harbor Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package jobmonitor
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/goharbor/harbor/src/lib/log"
+
+	"github.com/gocraft/work"
+
+	"github.com/goharbor/harbor/src/common/job"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/q"
+	libRedis "github.com/goharbor/harbor/src/lib/redis"
+	jm "github.com/goharbor/harbor/src/pkg/jobmonitor"
+	"github.com/goharbor/harbor/src/pkg/scheduler"
+	"github.com/goharbor/harbor/src/pkg/task"
+)
+
+// All the jobs in the pool, or all pools
+const All = "all"
+
+// Ctl the controller instance of the worker pool controller
+var Ctl = NewWorkerPoolController()
+
+// WorkerPoolController defines the worker pool operations
+type WorkerPoolController interface {
+	// ListPools lists the worker pools
+	ListPools(ctx context.Context) ([]*jm.WorkerPool, error)
+	// ListWorkers lists the workers in the pool
+	ListWorkers(ctx context.Context, poolID string) ([]*jm.Worker, error)
+	// StopRunningJob stop the running job
+	StopRunningJob(ctx context.Context, jobID string) error
+}
+
+type workerPoolController struct {
+	poolManager   jm.PoolManager
+	workerManager jm.WorkerManager
+	taskManager   task.Manager
+	sch           scheduler.Scheduler
+	monitorClient func() (jm.JobServiceMonitorClient, error)
+}
+
+// NewWorkerPoolController ...
+func NewWorkerPoolController() WorkerPoolController {
+	return &workerPoolController{
+		poolManager:   jm.NewPoolManager(),
+		workerManager: jm.NewWorkerManager(),
+		taskManager:   task.NewManager(),
+		monitorClient: jobServiceMonitorClient,
+	}
+}
+
+func (w *workerPoolController) StopRunningJob(ctx context.Context, jobID string) error {
+	if strings.EqualFold(jobID, All) {
+		allRunningJobs, err := w.allRunningJobs(ctx)
+		if err != nil {
+			log.Errorf("failed to get all running jobs: %v", err)
+			return err
+		}
+		for _, jobID := range allRunningJobs {
+			if err := w.StopRunningJob(ctx, jobID); err != nil {
+				log.Errorf("failed to stop running job %s: %v", jobID, err)
+				return err
+			}
+		}
+		return nil
+	}
+	tasks, err := w.taskManager.List(ctx, &q.Query{Keywords: q.KeyWords{"job_id": jobID}})
+	if err != nil {
+		return err
+	}
+	if len(tasks) == 0 {
+		return errors.NotFoundError(nil).WithMessage("job %s not found", jobID)
+	}
+	if len(tasks) != 1 {
```

Comment:
we would like to see 500 or 409 error code here?

Reply:
return 400